src/exporters/s5.py

```diff
@@ -0,0 +1,488 @@
+from pathlib import Path
+import itertools
+import numpy as np
+from pathos.pools import _ThreadPool as pool
+
+from typing import Dict, Optional, List
+from .all_valid_s5 import datasets as dataset_reference
+from .base import get_kenya
+from .cds import CDSExporter
+
+
+class S5Exporter(CDSExporter):
+    def __init__(
+        self,
+        pressure_level: bool,
+        granularity: str = "monthly",
+        data_folder: Path = Path("data"),
+        product_type: Optional[str] = None,
+        dataset: Optional[str] = None,
+    ) -> None:
+        """
+        Arguments
+        --------
+        data_folder: Path
+            path to the data folder (where data will be stored)
+
+        variable: str
+            the variable that you want to download.
+
+        pressure_level: bool
+            Do you want data at different atmospheric heights?
+            True = yes want data at different `pressure-levels`
+            False = no want only `single-level` data
+
+        granularity: str, {'monthly', 'hourly'}, default: 'monthly'
+            The granularity of data being pulled.
+            'hourly'/'pressure_level' data has a forecast for every 12hrs (12, 24, 36 ...)
+            'hourly'/'single_level' data has a forecast for every 6hrs (6, 12, 18, 24 ...)
+            'monthly' data has a forecast for every month {1 ... 6}
+
+        product_type : str
+            The product type is only valid for monthly datasets. It corresponds
+            to the post-processing of the monthly forecasts
+            Defaults to 'monthly_mean' if no product_type is given
+                if pressure_level == True:
+                 {'ensemble_mean', 'hindcast_climate_mean', 'monthly_mean'}
+                else
+                 {'ensemble_mean', 'hindcast_climate_mean', 'monthly_mean',
+                  'monthly_standard_deviation', 'monthly_maximum', 'monthly_minimum'}
+
+        Notes:
+        -----
+        - because the `dataset` argument is not intuitive, the s5 downloader
+        makes decisions for the user based on their preferences in the args
+        `pressure_level` and `granularity`.
+        - these are constant for one download
+        """
+        super().__init__(data_folder)
+
+        # initialise attributes for this export
+        self.pressure_level = pressure_level
+        self.granularity = granularity
+
+        if dataset is None:
+            self.dataset = self.get_dataset(self.granularity, self.pressure_level)
+        else:
+            self.dataset = dataset
+
+        # get the reference dictionary that corresponds to that dataset
+        self.dataset_reference = dataset_reference[self.dataset]
```

Comment:
@gabrieltseng why does this assume that the type is `Optional[Sequence[str]]` when 1) it's not optional 2) it's a dictionary 3) the values for the dictionary vary between `List`, `str`, `None`

```
src/exporters/s5.py:359: error: Incompatible return value type (got "Optional[Sequence[str]]", expected "List[Any]")
src/exporters/s5.py:388: error: Unsupported right operand type for in ("Optional[Sequence[str]]")
```

Reply:
Fixed by adding type hints